src/WPF/ArcGISRuntime.WPF.Samples/Samples/Data/EditAndSyncFeatures/EditAndSyncFeatures.xaml.cs

```diff
@@ -0,0 +1,511 @@
+                if (job.Error != null)
+                {
+                    message += ": " + job.Error.Message;
+                }
+                else
+                {
+                    // If no error, show messages from the job
+                    var m = from msg in job.Messages select msg.Message;
+                    message += ": " + string.Join<string>("\n", m);
+                }
+
+                // Show the message
+                ShowStatusMessage(message);
+            }
+        }
+
+        private void SyncGeodatabase()
+        {
+            // Return if not ready
+            if (_readyForEdits != EditState.Ready) { return; }
+
+            // Create parameters for the sync task
+            SyncGeodatabaseParameters parameters = new SyncGeodatabaseParameters()
+            {
+                GeodatabaseSyncDirection = SyncDirection.Bidirectional,
+                RollbackOnFailure = false
+            };
+
+            // Add each layer to the sync job
+            foreach (GeodatabaseFeatureTable table in _resultGdb.GeodatabaseFeatureTables)
+            {
+                // Get the ID for the layer
+                long id = table.ServiceLayerId;
+
+                // Create the SyncLayerOption
+                SyncLayerOption option = new SyncLayerOption(id);
+
+                // Add the option
+                parameters.LayerOptions.Add(option);
+            }
+
+            // Create job
+            SyncGeodatabaseJob job = _gdbSyncTask.SyncGeodatabase(parameters, _resultGdb);
+
+            // Subscribe to status updates
+            job.JobChanged += Job_JobChanged;
+
+            // Subscribe to progress updates
+            job.ProgressChanged += Job_ProgressChanged;
+
+            // Start the sync
+
+            job.Start();
+        }
+
+        private void Job_ProgressChanged(object sender, EventArgs e)
+        {
+            // Get the job object
+            SyncGeodatabaseJob job = sender as SyncGeodatabaseJob;
+
+            // Update the progress bar
+            UpdateProgressBar(job.Progress);
+        }
+
+        // Get the path to the tile package used for the basemap
+        private async Task<string> GetTpkPath()
+        {
+            // The desired tpk is expected to be called SanFrancisco.tpk
+            string filename = "SanFrancisco.tpk";
+
+            // The data manager provides a method to get the folder
+            string folder = DataManager.GetDataFolder();
+
+            // Get the full path
+            string filepath = Path.Combine(folder, "SampleData", "GenerateGeodatabase", filename);
+
+            // Check if the file exists
+            if (!File.Exists(filepath))
+            {
+                // Download the map package file
+                await DataManager.GetData("3f1bbf0ec70b409a975f5c91f363fe7d", "GenerateGeodatabase");
+            }
+            return filepath;
+        }
+
+        private string GetGdbPath()
```

Comment:
Why do we have this function (GetGdbPath)? It is only used once. Why not put the following line of code :
  return Environment.ExpandEnvironmentVariables("%TEMP%\\wildfire.geodatabase");
directly in the Initialize function?